src/domain/swap/swap-out-checker.ts

```diff
@@ -0,0 +1,40 @@
+import { toSats } from "@domain/bitcoin"
+
+import { SwapServiceError } from "./errors"
+
+export const SwapOutChecker = ({
+  minOnChainHotWalletBalanceConfig,
+  swapOutAmount,
+}: {
+  minOnChainHotWalletBalanceConfig: Satoshis
+  swapOutAmount: Satoshis
+}) => {
+  // checks the amount of "swap out" that is need
+  // if return "0" then no swap out needed
+  // if return an "amount" then swapout needed
+  // if return "error" then we don't have enough outbound
+  //   liquidity to perform a swap out.
+  const getSwapOutAmount = ({
+    currentOnChainHotWalletBalance,
+    currentOutboundLiquidityBalance,
+  }: {
+    currentOnChainHotWalletBalance: Satoshis
+    currentOutboundLiquidityBalance: Satoshis
+  }): Satoshis | SwapServiceError => {
```

Comment:
This function is in `domain` so it can not return `SwapServiceError` - as it is not a `service` function (ie. a function on `ISwapOutService`.

Also `Satoshis` is deprecated `BtcPaymentAmount` is the 'new' type for this.